living_documentation_generator/generator.py

```diff
@@ -140,73 +209,108 @@ def _fetch_github_project_issues(self) -> dict[str, ProjectIssue]:
+    ) -> dict[str, ConsolidatedIssue]:
+        """
+        Consolidate the fetched issues and extra project data into a one consolidated object.
+
+        Args:
+            repository_issues (dict): A dictionary containing repository issue objects with unique key.
+            projects_issues (dict): A dictionary containing project issue objects with unique key.
+        """
 
         consolidated_issues = {}
 
         # Create a ConsolidatedIssue object for each repository issue
         for repository_id in repository_issues.keys():
             for repository_issue in repository_issues[repository_id]:
                 repo_id_parts = repository_id.split("/")
-                unique_key = make_issue_key(repo_id_parts[0], repo_id_parts[1], repository_issue.number)
-                consolidated_issues[unique_key] = ConsolidatedIssue(repository_id=repository_id,
-                                                                    repository_issue=repository_issue)
+                unique_key = make_issue_key(
+                    repo_id_parts[0], repo_id_parts[1], repository_issue.number
+                )
+                consolidated_issues[unique_key] = ConsolidatedIssue(
+                    repository_id=repository_id, repository_issue=repository_issue
+                )
 
         # Update consolidated issue structures with project data
         logger.debug("Updating consolidated issue structure with project data.")
         for key, consolidated_issue in consolidated_issues.items():
             if key in projects_issues:
                 consolidated_issue.update_with_project_data(projects_issues[key].project_status)
 
-        logging.info("Issue and project data consolidation - consolidated `%s` repository issues"
-                     " with extra project data.",
-                     len(consolidated_issues))
+        logging.info(
+            "Issue and project data consolidation - consolidated `%s` repository issues"
+            " with extra project data.",
+            len(consolidated_issues),
+        )
         return consolidated_issues
 
-    def _generate_markdown_pages(self, issues: dict[str, ConsolidatedIssue]):
-        with open(LivingDocumentationGenerator.ISSUE_PAGE_TEMPLATE_FILE, 'r', encoding='utf-8') as f:
+    def _generate_markdown_pages(self, issues: dict[str, ConsolidatedIssue]) -> None:
+        """
+        Generate the Markdown pages for all consolidated issues, create a summary index page and
+        save it all to the output directory.
+
+        Args:
+            issues (dict): A dictionary containing all consolidated issues.
+        """
+        with open(LivingDocumentationGenerator.ISSUE_PAGE_TEMPLATE_FILE, "r", encoding="utf-8") as f:
             issue_page_detail_template = f.read()
 
-        with open(LivingDocumentationGenerator.INDEX_PAGE_TEMPLATE_FILE, 'r', encoding='utf-8') as f:
+        with open(LivingDocumentationGenerator.INDEX_PAGE_TEMPLATE_FILE, "r", encoding="utf-8") as f:
```

Comment:
regarding these 2 with statements: what if the file was not opened successfully for whatever reason? Then `issue_page_detail_template` and `issue_index_page_template` wouldn't be available and the program would crash. Perhaps consider to either move operations working with them into the individual `with` scope, or initialize the variable above as `None`

Reply:
Logic was updated at: [5db1778](https://github.com/AbsaOSS/living-doc-generator/pull/30/commits/5db1778f2ea5b37dd99c4b062d94c7c3ed9eded2)